src/levanter/data/loader.py

```diff
@@ -61,40 +100,51 @@ def __init__(
         self.axis_resources = axis_resources
         self.data_store = data
         self.mesh = mesh
-        self.Batch = Batch
+
+        if batch_size is None:
+            assert isinstance(batch_axis_name, hax.Axis)
+            self.batch_axis_name = batch_axis_name.name
```

Comment:
This is what led me to the comment above - just feels yucky enough to do `...name.name` that it could possibly be confusing as well down the line.